### PR TITLE
fix(gatsby): Throw Typegen errors & add `IGatsbyImageData` to output

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

- Make two GraphQL errors less verbose as they are thrown when there is also a typegen error. This way the terminal overview is better
- Throw proper structured error for typegen errors
- End activity and don't `panic` during develop
- Always use validation for documents to get proper error messages
- Update flag description and not mark as experimental anymore
- Use `IGatsbyImageData` type for gatsbyImageData

### Documentation

I'll add a `docsUrl` to the error in https://github.com/gatsbyjs/gatsby/pull/35584

## Related Issues

[ch50768]
[ch50542]
https://github.com/gatsbyjs/gatsby/discussions/35420#discussioncomment-2732724

### CODE REVIEW

**Comment by aaronadamsCA** [comment]
Oh good, I was just coincidentally checking how you worked around the generic `JSON` scalar type.

Related feature request: #33317

**Comment by aaronadamsCA** [comment]
Also, if this helps, in our environment the `gatsby-plugin-image#IGatsbyImageData` syntax didn't work. We had to use a dynamic import expression (`import('gatsby-plugin-image').IGatsbyImageData`) instead.

**Comment by LekoArts** [comment]
> Also, if this helps, in our environment the `gatsby-plugin-image#IGatsbyImageData` syntax didn't work. We had to use a dynamic import expression (`import('gatsby-plugin-image').IGatsbyImageData`) instead.

Works in my testing. It adds

```ts
import type { IGatsbyImageData } from 'gatsby-plugin-image';
```

At the top of the file and then reuses that in the scalars

```ts
type Scalars = {
  GatsbyImageData: IGatsbyImageData;
  JSON: Record<string, unknown>;
};
```

**Comment by LekoArts** [comment]
Oh, you were right. The global namespace doesn't work if you add an import, so I switched it to the dynamic import.

Luckily this was also caught by the e2e tests 👍

**Comment by aaronadamsCA** [comment]
@LekoArts That was the only one we had encountered so far.

Looking forward to seeing this ship!

**Comment by wardpeet** [comment]
Looks good, do you want to do the same for imageCDN located in gatsby-plugin-utils?

**Review by wardpeet — APPROVED** [review]

File: packages/gatsby-plugin-utils/src/polyfill-remote-file/graphql/gatsby-image-resolver.ts [hunk 1]
```diff
@@ -260,7 +260,7 @@ export function generateGatsbyImageFieldConfig(
   IGatsbyImageDataArgs
 > {
   return {
-    type: `JSON`,
+    type: `GatsbyImageData!`,
```
Comment (pieh) [inline packages/gatsby-plugin-utils/src/polyfill-remote-file/graphql/gatsby-image-resolver.ts:263]: This is a breaking change in sense that it requires users to bump `gatsby` package (to version that actually creates that scalar) (this comment apply to all places that make similar swap)

**Review by pieh — CHANGES_REQUESTED** [review]

**Review by pieh — APPROVED** [review]

**Pull request merged**